os/fs/smartfs/smartfs_utils.c

```diff
@@ -1361,15 +1358,13 @@ int smartfs_writeentry(struct smartfs_mountpt_s *fs, struct smartfs_entry_s new_
 #endif
 	direntry->datlen = 0;
 	if (direntry->name == NULL) {
-		direntry->name = (FAR char *)kmm_malloc(fs->fs_llformat.namesize + 1);
+		direntry->name = (FAR char *)kmm_zalloc(fs->fs_llformat.namesize + 1);
```

Comment:
this direntry is useless as I told you many time,

Reply:
You mean there's no need for allocating name space for diretnry here?
Or no need to return updated entry in direntry at all?

Reply:
2nd one.